langlib/jballerina.java/src/main/ballerina/annotations.bal

```diff
@@ -126,3 +126,13 @@ public const annotation FieldData FieldSet on source external;
 # }
 # ```
 public const annotation ObjectData Binding on class;
+
+# Describes a type, class or function that is a dependency of java native code. This annotation should be added by the
+# developer if a type definition is being accessed by a `io.ballerina.runtime.api.creators` api such as `ValueCreator`.
+# ```ballerina
+# @ExternalDependency
+# function foo(int x) returns int {
+#     return x * 2;
+# }
+# ```
+public const annotation ExternalDependency on source type, source class, source function;
```

Comment:
You are correct. I checked with @HindujaB from runtime team and according to her, currently there is no API to invoke functions by providing their names.

However, the runtime team is working on such a feature and it will be added in a future update. Furthermore, even though it is highly discouraged, users can invoke ballerina functions from native code using Java reflection API. Therefore, adding support for functions will be future proof in my opinion.